pkg/debuginfo/debuginfo.go

```diff
@@ -347,49 +359,49 @@ func hasDebugInfo(path string) (bool, error) {
 }
 
 func hasDWARF(path string) (bool, error) {
-	exe, err := elf.Open(path)
+	f, err := elf.Open(path)
 	if err != nil {
 		return false, fmt.Errorf("failed to open elf: %w", err)
 	}
-	defer exe.Close()
+	defer f.Close()
 
-	data, err := getDWARF(exe)
+	data, err := dwarf(f)
 	if err != nil {
 		return false, fmt.Errorf("failed to read DWARF sections: %w", err)
 	}
 
 	return len(data) > 0, nil
 }
 
-// A simplified and modified version of debug/elf.DWARF().
-func getDWARF(f *elf.File) (map[string][]byte, error) {
-	dwarfSuffix := func(s *elf.Section) string {
-		switch {
-		case strings.HasPrefix(s.Name, ".debug_"):
-			return s.Name[7:]
-		case strings.HasPrefix(s.Name, ".zdebug_"):
-			return s.Name[8:]
-		case strings.HasPrefix(s.Name, "__debug_"): // macos
-			return s.Name[8:]
-		default:
-			return ""
-		}
+var dwarfSuffix = func(s *elf.Section) string {
```

Comment:
nit: I'm being incredibly pedantic here but this should be `dwarfPrefix` as opposed to `dwarfSuffix`.

Reply:
This is actually, removing the detected prefix and returning the suffix. Thus the name. Maybe I should a comment to clarify more.

Reply:
Aah, ok I understand now. Sorry for the noise!